main/ble2mqtt.c

```diff
@@ -43,6 +42,11 @@ static char *device_name_get(void)
             mac = eth_mac_get();
             break;
         case NETWORK_TYPE_WIFI:
+            if (config_wifi_hostname_get() != NULL)
+            {
+                sprintf(name, config_wifi_hostname_get());
```

Comment:
It's a bit dangerous to do this as the string in the configuration file may be longer that that of `name`. You can just return `config_wifi_hostname_get()` directly as it's a pointer that should remain valid.

Reply:
True, what i wanted to accomplish here is that it should take the config_wifi_hostname_get() if it is entered, otherwise still default back to the mac part which is just below this one. How would i tackle this then?

Reply:
> Hey,
> 
> Thanks for the pull request!
> In this commit you also move the configuration file to a sample. I understand the point of it and considered doing something similar myself but it should be a separate commit/PR.

Agree, reverted that part.

> As for the hostname, I suggest changing the configuration file to include a general `network` field and move `wifi` and `eth` to be inside it. The new `hostname` will also be under that `network` field and will leave room for future options like configuring static IP, DNS and the likes.

Agree, if you answer below i will also implement it as such.

Reply:
> It's a bit dangerous to do this as the string in the configuration file may be longer that that of `name`. You can just return `config_wifi_hostname_get()` directly as it's a pointer that should remain valid.

Fixed (can you confirm)

Reply:
You should change the definition of `device_name_get()` to
```c
static const char *device_name_get(void)
```
No one should change the return valued of this function.